src/components/ContentHighlights/ContentHighlightArchivedAlert.jsx

```diff
@@ -0,0 +1,49 @@
+import { Alert } from '@edx/paragon';
+import { useContext, useState } from 'react';
+import { NEW_ARCHIVED_CONTENT_ALERT_DISMISSED_COOKIE_NAME } from './data/constants';
+import { EnterpriseAppContext } from '../EnterpriseApp/EnterpriseAppContextProvider';
+import { enterpriseCurationActions, isArchivedContent } from '../EnterpriseApp/data/enterpriseCurationReducer';
+
+const ContentHighlightArchivedAlert = () => {
+  const archivedContentCookies = {};
+  const [isOpen, setIsOpen] = useState(true);
+
+  const { enterpriseCuration: { enterpriseHighlightedSets, dispatch } } = useContext(EnterpriseAppContext);
+  enterpriseHighlightedSets.forEach(set => {
+    set.highlightedContent.forEach(content => {
+      if (isArchivedContent(content)) {
+        if (archivedContentCookies[set.uuid]) {
+          archivedContentCookies[set.uuid].push(content.contentKey);
+        } else {
+          archivedContentCookies[set.uuid] = [content.contentKey];
+        }
+      }
+    });
+  });
```

Comment:
[nit/suggestion] I believe this logic will be executed on every re-render, which could be a bit expensive given it's iterating through multiple `forEach` loops.

Would it make sense to ensure this logic is only executed when necessary (when data changes) vs. on every re-render?

```jsx
const [archivedContentCookies, setArchivedContentCookies] = useState({});

const { enterpriseCuration: { enterpriseHighlightedSets, dispatch } } = useContext(EnterpriseAppContext);

useEffect(() => {
    enterpriseHighlightedSets.forEach((highlightSet) => {
      highlightSet.highlightedContent.forEach((content) => {
        if (!isArchivedContent(content)) {
          // content not archived; do nothing.
          return;
        }
        // content is archived, so update `archivedContentCookies` state.
        setArchivedContentCookies((prevState) => {
          if (!prevState[highlightSet.uuid]) {
            prevState[highlightSet.uuid] = [content.contentKey];
          } else {
            prevState[highlightSet.uuid].push(content.contentKey);
          }
          return prevState;
        });
      });
    });
}, [enterpriseHighlightedSets]);
```